phoenix-core/src/main/java/org/apache/calcite/jdbc/PhoenixCalciteFactory.java

```diff
@@ -161,6 +162,29 @@ public Object apply(
             }));
         }
 
+        public CalciteStatement createStatement(String sql, int resultSetType, int resultSetConcurrency, int resultSetHoldability) throws SQLException {
+            try {
+                return super.createStatement(resultSetType, resultSetConcurrency, resultSetHoldability);
+            } catch (SQLException e) {
+                if (e.getCause().getCause() instanceof SQLException) {
+                    throw (SQLException) e.getCause().getCause();
```

Comment:
The new changes go for the deepest SQL exception. Can quickly change it if you would like only the next most deep SQL exception.